### ISSUE TITLE
Content Type Not copied

### ISSUE THREAD

**Issue by abulhassanoasis** [Zerg00s/FlowPowerAppsMigrator#15]
Hi

When we execute the 'RunAllScripts.ps1. file, and mention the source & destination lists, the program doesn't copy the content type associated with the source list across to the destination list. 

Can you please advise if this feature is not included in the migrator? 

Thanks a lot!

**Comment by Zerg00s** [comment]
hi @abulhassanoasis,

Content types are not migrated by this tool, unfortunately.

**Comment by KBolland** [comment]
Hi @Zerg00s,

Just picking up on the above and my previous post in the SharePoint list copying.

We are have are now getting the issue saying that the content type ID is not available on the target site when the copy happens. I understand from the above that the solution doesn't copy the content type. But if the content type was setup already and we didn't copy the lists and libraries would your solution work?

I thought it might be useful to explain what we are trying to achieve with your solution to give you some context.

We have a live production site which contains a document library with Power App SharePoint List form. We are now creating a mirror development site and document library including the Power App SharePoint List Form. So what we hope to do with your application is firstly copy the Power Apps SharePoint list form from the current document library to the mirror document library on the development site.

Once completed we hope then to use your tool update the Power App SharePoint list form on the Live production site from the document library on the development site.

I hope that helps and you are able to help.

Thanks again for the great solution!

Karl

**Comment by KBolland** [comment]
Thanks @Zerg00s for getting back to me so quickly.

I have had a look at the MS lists app but it doesn't appear to pickup document libraries for me.

The scenario is we have a document library using a content type that was created on the site. The document library then has a custom PowerApp SP form setup on it. We are now trying to create a mirror of the site and importantly the document library and PowerApp SP form to allow us to develop without affecting the live document library.

We are currently looking into the new Content type hub site to see if that helps us as we are hoping it will keep the same ID.

Have you had any experience of this?

Many thanks.

**Comment by Zerg00s** [comment]
hi @KBolland,

Just wanted to post an update: I'm working on a new release that will migrate Content Types as well. I can't promise it, but it should be ready within the next 2 days. I also think, the current version is not very stable. I am having difficulties migration List Item forms customized with Power Apps. It used to work fine, but now it does not seem to work properly. 

As to the Content Type hub, I haven't tried the new one, but I am guessing it works similar to the old one. It's a good option, of course, but you will still need to migrate lists. 

In general, I am trying to avoid content types when working with Power Apps and Power Automate. Instead, I create all columns on a list level.

**Comment by Zerg00s** [comment]
hi @KBolland,

I've added an ability to move content types to the [Move-Lists tool](https://github.com/Zerg00s/Move-Lists). Maybe you can give it a try? 

In the meanwhile, I am still working on adding the same feature to the Flow & app migrator.

**Comment by KBolland** [comment]
Hi @Zerg00s ,

The library copy worked perfectly as far as I can see with the content type included so thank you so much for doing that so quickly.

After having the library copied I ran the PowerApps migrator and all looked to complete. But then when I went to the library it and view the form it appeared to load the form that was on the original site not the target site. I then review the form settings on the library and it shows 'Use the default SharePoint form' as the selected form option on the form settings.

Not sure if I missed a step or something but are you able to advise?

Many thanks.

**Comment by Zerg00s** [comment]
Hi @KBolland ,

I am experiencing the exact same issue. In the past, around 5-6 weeks ago I was able to migrate SharePoint forms customized with Power Apps, but as of today, after migration I see that the original form is still used. In the list settings I see: the `use the default SharePoint form` option selected. 

As an experiment, I tried customizing the form with Power apps on the target site. And it actually opens my migrated Power App form in Edit mode.  But when I save and publish it, the target SharePoint list still shows the original SharePoint form.

I hope it's just a bug in my code, but I am afraid it might be due to some change to the SharePoint or Power Apps API. I am trying to figure out how to fix it.
